Mathlib/Topology/DiscreteSubset.lean

```diff
@@ -98,6 +104,32 @@ theorem isClosed_and_discrete_iff {S : Set X} :
   · refine ⟨fun hx ↦ ?_, fun _ ↦ H⟩
     simpa [disjoint_iff, nhdsWithin, inf_assoc, hx] using H
 
+/-- The union of two discrete closed subsets is discrete. -/
+theorem discreteTopology_union {S T : Set X} (hs : DiscreteTopology S) (ht : DiscreteTopology T)
+    (hs' : IsClosed S) (ht' : IsClosed T) : DiscreteTopology ↑(S ∪ T) := by
+  refine (isClosed_and_discrete_iff.mpr fun x ↦ ?_).2
+  exact Filter.sup_principal ▸ (isClosed_and_discrete_iff.mp ⟨hs', hs⟩ x).sup_right
+    (isClosed_and_discrete_iff.mp ⟨ht', ht⟩ x)
+
+/-- The union of finitely many discrete closed subsets is discrete. -/
+theorem discreteTopology_biUnion_finset {ι : Type*} {I : Finset ι} {s : ι → Set X}
+    (hs : ∀ i ∈ I, DiscreteTopology (s i)) (hs' : ∀ i ∈ I, IsClosed (s i)) :
+    DiscreteTopology (⋃ i ∈ I, s i) := by
+  classical
+  induction I using Finset.induction_on with
+  | empty => rw [show ⋃ i ∈ ∅, s i = ∅ by simp]; infer_instance
```

Comment:
This feels like it should be a single lemma too, but it seems to be missing. Bonus points if you add it :)

Reply:
You mean the thing inside the brackets? The lemma seems to be 
```lean
theorem Set.biUnion_empty_finset {ι X : Type*} {s : ι → Set X} :
    ⋃ i ∈ (∅ : Finset ι), s i = ∅ := by
  simp only [Finset.notMem_empty, iUnion_of_empty, iUnion_empty]
```
and then
`| empty => rw [Set.biUnion_empty_finset]; infer_instance`
works. 

The strange thing is that `Set.biUnion_empty_finset` cannot be a simp lemma, because simp can already prove it; but using the same three simp lemmas directly, as
`| empty => simp only [Finset.notMem_empty, iUnion_of_empty, iUnion_empty]; infer_instance`
doesn't work. Maybe `simp` is unhappy rewriting inside terms coerced to subtypes?